src/silo_api/request_handler_factory.test.cpp

```diff
@@ -22,28 +22,32 @@ class MockDatabase : public silo::Database {
    MOCK_METHOD(silo::DatabaseInfo, getDatabaseInfo, (), (const));
    MOCK_METHOD(silo::DetailedDatabaseInfo, detailedDatabaseInfo, (), (const));
    MOCK_METHOD(silo::DataVersion, getDataVersion, (), (const));
+
+   MOCK_METHOD(silo::query_engine::QueryResult, executeQuery, (const std::string&), (const));
 };
 
-class MockQueryEngine : public silo::query_engine::QueryEngine {
+class MockDatabaseMutex : public silo_api::DatabaseMutex {
   public:
-   explicit MockQueryEngine(const silo::Database& database)
-       : QueryEngine(database) {}
+   std::shared_mutex mutex;
+   MockDatabase mock_database;
 
-   MOCK_METHOD(silo::query_engine::QueryResult, executeQuery, (const std::string&), (const));
+   silo_api::FixedDatabase getDatabase() override {
+      std::shared_lock<std::shared_mutex> lock(mutex);
+      return silo_api::FixedDatabase(mock_database, std::move(lock));
```

Comment:
"Clang-Tidy: Avoid repeating the return type from the declaration; use a braced initializer list instead"